src/main/java/net/neoforged/neoforge/transfer/access/ItemAccess.java

```diff
@@ -0,0 +1,207 @@
+     * Any {@linkplain #insert insertion} is always accepted, but only one item is actually added
+     * to the player's inventory, and only if the player does not already have the item.
+     * This matches the behavior of {@link ItemUtils#createFilledResult}.
+     */
+    static ItemAccess forInfiniteMaterials(Player player, ItemStack contents) {
+        return new InfiniteMaterialsItemAccess(player, ItemResource.of(contents), contents.getCount());
+    }
+
+    /**
+     * Creates an item access instance for a player's cursor in a menu.
+     */
+    static ItemAccess forPlayerCursor(Player player, AbstractContainerMenu menu) {
+        return new PlayerItemAccess(PlayerInventoryWrapper.of(player), CarriedSlotWrapper.of(menu));
+    }
+
+    /**
+     * Creates an item access instance for a specific slot of a player.
+     */
+    static ItemAccess forPlayerSlot(Player player, int slot) {
+        var inventoryWrapper = PlayerInventoryWrapper.of(player);
+        return new PlayerItemAccess(inventoryWrapper, RangedResourceHandler.ofSingleIndex(inventoryWrapper, slot));
+    }
+
+    /**
+     * Creates an item access instance for a specific slot of an item resource handler.
+     *
+     * <p>Overflow on insertion will be sent to the rest of the handler via
+     * {@linkplain ResourceHandler#insert(IResource, int, TransactionContext) the slotless insert} method.
+     * If this is not desired, create a single slot wrapper using {@link RangedResourceHandler#ofSingleIndex}.
+     */
+    static ItemAccess forHandlerIndex(ResourceHandler<ItemResource> handler, int index) {
+        return new HandlerItemAccess(handler, index);
+    }
+
+    /**
+     * Creates an item access instance that will mutate a stack directly,
+     * possibly changing the components and the count, but never the underlying Item as it's final.
+     *
+     * <p>This can be used when it is known that the underlying Item will not change.
+     *
+     * @throws IllegalArgumentException if the stack is empty
+     */
+    static ItemAccess forStack(ItemStack stack) {
+        if (stack.isEmpty()) {
+            throw new IllegalArgumentException("Expected stack to be non-empty.");
+        }
+        return new StackItemAccess(stack);
+    }
+
+    /**
+     * Creates a wrapper around this access that allows access to a single item at the time.
+     */
+    @ApiStatus.NonExtendable
+    default ItemAccess oneByOne() {
+        return new OneByOneItemAccess(this);
+    }
+
+    /**
+     * Retrieves a capability from this item location.
+     *
+     * @implNote This method is a convenient shorthand for {@code location.getResource().toStack().getCapability(capability, location)}.
+     */
+    @Nullable
+    @ApiStatus.NonExtendable
+    default <T> T getCapability(ItemCapability<T, ItemAccess> capability) {
+        return getResource().toStack().getCapability(capability, this);
```

Comment:
Instead of the extra redirection which might be annoying to have to do an extra step through when going with a debugger, can we just have this directly call the method? Aka:
```suggestion
        return capability.getCapability(getResource().toStack(), this);
```